dashboard/app/controllers/sections_controller.rb

```diff
@@ -1,10 +1,20 @@
 class SectionsController < ApplicationController
-  before_action :load_section
+  before_action :load_section_by_code, except: :update
 
   def show
     @secret_pictures = SecretPicture.all.shuffle
   end
 
+  # Allows you to update a section's course_id. Clears any assigned script_id
+  # in the process
+  def update
```

Comment:
This breaks my mental model of `update` updating only the things that I ask to update. Does it make sense to name this `update_course_id_clear_script_id`? Does it make sense to make callers of this pass in `script_id: nil`?